axlearn/common/launch.py

```diff
@@ -13,6 +13,11 @@
     "--xla_tpu_spmd_rng_bit_generator_unsafe=1",  # SPMD partition-aware RngBitGenerator.
     "--xla_tpu_enable_latency_hiding_scheduler=true",  # Try to schedule ops efficiently.
     "--xla_tpu_perform_spmd_cse_prevention=false",  # b/229655601: prevent OOM on gpt2-small-repeat.
+    # If MegaScale Runtime Error is encountered when running multi-slice jobs,
+    # enabling this flag will allow for termination of the job, triggering
+    # auto-restart of the nodes. This is set to true to prevent the job from
+    # silently hanging and to reduce time to recovery.
+    "--megascale_abort_on_hangs=true"
```

Comment:
does this work on all tpu versions?

Reply:
--megascale_abort_on_hangs is platform agnostic so it works for all TPU versions.